copycat/runtime/src/main/java/org/apache/kafka/copycat/runtime/WorkerSinkTask.java

```diff
@@ -108,6 +108,9 @@ public void close() {
     /** Poll for new messages with the given timeout. Should only be invoked by the worker thread. */
     public void poll(long timeoutMs) {
         try {
+            rewind();
+            long backOffMs = context.backoff();
+            timeoutMs = Math.max(timeoutMs, backOffMs);
```

Comment:
Also, don't we either need to record the starting time for the backoff or reset it at some point? Otherwise we'll keep using the timeout indefinitely. Or was the expectation that the task reset it when it no longer needs it?